content/en/blog/2024/otel-errors/index.md

```diff
@@ -0,0 +1,286 @@
+flexibility for each language to implement features as idiomatically as possible. 
+For example, most languages have implemented `RecordException`, while Go has 
+implemented `RecordError`, which does the same thing. 
+
+You can view this [compliance matrix](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) of the spec across all languages, but you’ll 
+get the most updated info by checking the individual language repo. Now we have 
+a place from which to begin figuring out how to handle errors in OTel, starting 
+with how to report them:
+* In spans
+* In logs
+  
+### Errors Depicted in Spans
+In OTel, spans are the building blocks of distributed traces, representing 
+individual units of work within a distributed system. Spans are related to each 
+other and to a trace through context. Put simply, context is the glue that turns 
+a pack of data into a unified trace. Context propagation allows us to pass 
+information across multiple systems, therefore tying them together. Traces can 
+tell us all sorts of things about our applications through metadata and span 
+events.
+
+![Graphic that shows the spans within a trace](OTel-spans.png)
+
+### Enhancing Spans with Metadata
+OTel enables you to enhance spans with metadata ([attributes](https://opentelemetry.io/docs/concepts/signals/traces/#attributes)) in the form of key/value pairs. By attaching relevant 
+information to spans, such as user IDs, request parameters, or environment 
+variables, you can gain deeper insights into the circumstances surrounding an 
+error and quickly identify its root cause. This metadata-rich approach to error 
+handling can significantly reduce the time and effort required to diagnose and 
+resolve issues, ultimately improving the reliability and maintainability of your 
+applications.
+
+Spans also have a [span kind](https://opentelemetry.io/docs/concepts/signals/traces/#span-kind) 
+field, which gives us some additional metadata that can help developers 
+troubleshoot errors. OTel defines several span kinds, each of which has unique 
+implications for error reporting: 
+* **client**: For outgoing synchronous remote calls (e.g. outgoing HTTP request 
+or DB call)
+* **server**: For incoming synchronous remote calls (e.g. incoming HTTP request 
+or remote procedure call)
+* **internal**: For operations that do not cross process boundaries (e.g. 
+instrumenting a function call)
+* **producer**: For the creation of a job which may be asynchronously processed 
+later (e.g. job inserted into a job queue)
+* **consumer**: For the processing of a job created by a producer, which may start 
+long after the producer span has ended 
+
+Span kind is determined automatically by the instrumentation libraries used. In 
+the event that a span kind has been incorrectly set, you may be able to 
+[override](https://developer.newrelic.com/collect-data/opentelemetry-manual/custom-span-event/) 
```

Comment:
```suggestion
[override](https://developer.newrelic.com/collect-data/opentelemetry-manual/custom-span-event/) 
```

Can you update this with a link to the OpenTelemetry documentation?

Reply:
A bit weird... I know you were able to do this at one point -- I've done it before -- but now I cannot find any documentation on it. I see in [OTel-JS doc](https://open-telemetry.github.io/opentelemetry-js/classes/_opentelemetry_sdk_trace_base.Span.html#kind) it says span kind is a read-only property. So perhaps something changed. I'll remove this entire paragraph.